zone/client.cpp

```diff
@@ -2783,18 +2783,64 @@ void Client::GMKill() {
 	safe_delete(outapp);
 }
 
-void Client::MemorizeSpell(uint32 slot,uint32 spellid,uint32 scribing, uint32 reduction){
-	if (slot < 0 || slot >= EQ::spells::DynamicLookup(ClientVersion(), GetGM())->SpellbookSize)
+void Client::MemorizeSpell(uint32 slot, uint32 spell_id, uint32 scribing, uint32 reduction){
+	if (
+		!EQ::ValueWithin(
+			slot,
+			0,
+			(EQ::spells::DynamicLookup(ClientVersion(), GetGM())->SpellbookSize - 1)
```

Comment:
Why is this newly `- 1` versus how it was before?

ValueWithin does slot >= SpellbookSize

```
bool ValueWithin(const T &value, const T &lower, const T &upper)
{
    return value >= lower && value <= upper;
}
```

Reply:
We were returning if >= this value so we check if it’s not between 0 and 1 less than this value then return if it isn’t.